monai/data/wsi_reader.py

```diff
@@ -744,6 +763,101 @@ def get_mpp(self, wsi, level: int) -> tuple[float, float]:
+        lower_bound_y = mpp_closest_lvl_y * (1 - rtol) - atol
+        upper_bound_y = mpp_closest_lvl_y * (1 + rtol) + atol
+
+        # Check if user-provided mpp_x and mpp_y fall within the tolerance intervals for closest level
+        within_tolerance_x = (user_mpp_x >= lower_bound_x) & (user_mpp_x <= upper_bound_x)
+        within_tolerance_y = (user_mpp_y >= lower_bound_y) & (user_mpp_y <= upper_bound_y)
+        within_tolerance = within_tolerance_x & within_tolerance_y
+
+        if within_tolerance:
+            # Take closest_level and continue with returning img at level
+            print(f"User-provided MPP lies within tolerance of level {closest_lvl}, returning wsi at this level.")
+            closest_lvl_wsi = wsi.read_region(
+                (0, 0), level=closest_lvl, size=closest_lvl_dim, num_workers=self.num_workers
+            )
+
+        else:
+            # If mpp_closest_level < mpp -> closest_level has higher res than img at mpp => downscale from closest_level to mpp
+            closest_level_is_bigger_x = mpp_closest_lvl_x < user_mpp_x
+            closest_level_is_bigger_y = mpp_closest_lvl_y < user_mpp_y
+            closest_level_is_bigger = closest_level_is_bigger_x & closest_level_is_bigger_y
+
+            if closest_level_is_bigger:
+                ds_factor_x = mpp_closest_lvl_x / user_mpp_x
+                ds_factor_y = mpp_closest_lvl_y / user_mpp_y
+
+                closest_lvl_wsi = wsi.read_region(
+                    (0, 0), level=closest_lvl, size=closest_lvl_dim, num_workers=self.num_workers
+                )
+                wsi_arr = cp.array(closest_lvl_wsi)
+
+                target_res_x = int(np.round(closest_lvl_dim[1] * ds_factor_x))
+                target_res_y = int(np.round(closest_lvl_dim[0] * ds_factor_y))
+
+                closest_lvl_wsi = cucim_resize(wsi_arr, (target_res_x, target_res_y), order=0)
+                print(f"Case 1: Downscaling using factor {(ds_factor_x, ds_factor_y)}")
+
+            else:
+                # Else: increase resolution (ie, decrement level) and then downsample
+                closest_lvl = closest_lvl - 1
+                mpp_closest_lvl = mpp_list[closest_lvl]
+                mpp_closest_lvl_x, mpp_closest_lvl_y = mpp_closest_lvl
+
+                ds_factor_x = mpp_closest_lvl_x / user_mpp_x
+                ds_factor_y = mpp_closest_lvl_y / user_mpp_y
+
+                closest_lvl_dim = wsi.resolutions["level_dimensions"][closest_lvl]
+                closest_lvl_wsi = wsi.read_region(
+                    (0, 0), level=closest_lvl, size=closest_lvl_dim, num_workers=self.num_workers
+                )
+                wsi_arr = cp.array(closest_lvl_wsi)
+
+                target_res_x = int(np.round(closest_lvl_dim[1] * ds_factor_x))
+                target_res_y = int(np.round(closest_lvl_dim[0] * ds_factor_y))
+
+                closest_lvl_wsi = cucim_resize(wsi_arr, (target_res_x, target_res_y), order=0)
```

Comment:
I think order=0 equates to nearest neighbour interpolation. The other methods seem to be using bilinear interpolation (order=1). If so, this would seem to be preferable. An alternative would be to have an optional parameter - possibly overkill.

Reply:
Ah yes, you're right. That should be order=1 here. Thanks!
If wanted, I could keep bilinear interpolation as default, and add an optional parameter to enable the user to overwrite the default. Potentially, the user might also want to decide on which resize method to use (currently `cucim.skimage.transform.resize` and `PIL.Image.resize`), which could be another optional parameter.